dace/sourcemap.py

```diff
@@ -176,34 +177,27 @@ def create_py_map(sdfg):
         py_mapper.mapper(sdfg, tmp)
         folder = sdfg.build_folder
         save("py", sdfg.name, py_mapper.map, folder)
+        if tmp:
+            remove_tmp(sdfg.name, True)
         # If the SDFG was made with the API we need to create tmp info
         # as it doesn't have any
         sourceFiles = [src for src in get_src_files(sdfg, set())]
         # If tmp is None, then the SDFG was created with the API
-        if tmp is None:
-            temporaryInfo(
-                sdfg.name, {
-                    'build_folder': folder,
-                    'src_files': sourceFiles,
-                    'made_with_api': True,
-                })
-        else:
-            tmp['src_files'] = sourceFiles
-            temporaryInfo(sdfg.name, tmp)
+        return {
+            'build_folder': folder,
+            'src_files': sourceFiles,
+            'made_with_api': True if tmp is None else False,
+        }
```

Comment:
now that I think of it, this is a very javascript thing to do. In Python one would normally make an object (dataclass in this case would make a lot of sense)